core/services/log/broadcaster.go

```diff
@@ -237,7 +238,15 @@ func (b *broadcaster) startResubscribeLoop() {
 			return
 		}
 
-		chBackfilledLogs, abort := b.ethSubscriber.backfillLogs(b.latestHeadFromDb, addresses, topics)
+		var backfillFrom null.Int64
+		if b.latestHeadFromDb != nil {
+			// the backfill needs to start at an earlier block than the one last saved in DB,
+			// because any log is sent after a delay since the highest seen head -
+			// so at node re-start we need to account for those not-sent-yet logs that were in the buffer
```

Comment:
This comment could be clearer - the issue is that saving heads is not synchronized with the log broadcaster saving logs, right? So we need to "overshoot" on the heads to be sure that it's covered

Reply:
true.. i'm using BlockBackfillDepth now - on slower chains it will overshoot a bit too much, but at least it's configurable